### ISSUE TITLE
Static closure $this

### ISSUE THREAD

**Issue by fruitl00p** [kalessil/phpinspectionsea#1279]
I know that closures that don't mention `$this` in the function can already be optimized into a static closure. Thanks for that!

The problem is that when you add `$this` later on (i.e. in a refactor of the closure inside a method) PHPStorm doesn't complain about the undefined `$this`

(these are ugly codes, but to illustrate)

for example:
```php
class Foo
{
    public function bar()
    {
        return array_reduce([1, 2], function($carry, $i) {
            return $carry + $i;
        }, 0);
    }
    // some other method down the line
    public function add($a, $b)
    {
        return $a + $b;        
    }
}
```
would be refactored into:
```php
class Foo
{
    public function bar()
    {
        return array_reduce([1, 2], static function($carry, $i) {
            return $carry + $i;
        }, 0);
    }
    // some other method down the line
    public function add($a, $b)
    {
        return $a + $b;        
    }
}
```
BUT after yet another refactoring to utilize the 'add' method:
```php
class Foo
{
    public function bar()
    {
        return array_reduce([1, 2], static function($carry, $i) {
            return $this->add($carry, $i);
        }, 0);
    }
    // some other method down the line
    public function add($a, $b)
    {
        return $a + $b;
    }
}
```
PHPStorm wont complain... Is that a PHPstorm issue or could / should that be part of PHPInspectionsEA?

**Comment by kalessil** [comment]
@fruitl00p: this has been covered in Php Inspections (EA Ultimate) already.
